### PR TITLE
Action to run in PRs

### PR DESCRIPTION
**Pull request by jabolina**
Just a simple action, running the verify for JDK 17. For the future we should set up something using testcontainers or something similar.

### CODE REVIEW

File: .github/workflows/pull_requests.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: JDK 17 Tests
+        run: mvn -B verify -Dmaven.test.failure.ignore=true -Dansi.strip=true
+
+      - name: Test Logs
```
Comment (ryanemerson) [inline .github/workflows/pull_requests.yml:27]: In order for these results to be displayed you also need the test report action:

https://github.com/infinispan/protostream/blob/main/.github/workflows/test_report.yaml

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -2,25 +2,25 @@ quarkus.banner.enabled=false
 quarkus.package.type=uber-jar
 
 # Debezium configuration
-gingersnap.rule.us-east.connector.connector=MYSQL
-gingersnap.rule.us-east.connector.schema=debezium
-gingersnap.rule.us-east.connector.table=customer
+%dev.gingersnap.rule.us-east.connector.connector=MYSQL
```
Comment (ryanemerson) [inline src/main/resources/application.properties:5]: Why are the property changes required?

File: src/test/java/io/gingersnap_project/GreetingResourceTest.java [hunk 3]
```diff
@@ -1,11 +1,11 @@
 package io.gingersnap_project;
 
-import io.quarkus.test.junit.QuarkusTest;
-import org.junit.jupiter.api.Test;
-
 import static io.restassured.RestAssured.given;
 import static org.hamcrest.CoreMatchers.is;
 
+import io.quarkus.test.junit.QuarkusTest;
```
Comment (ryanemerson) [inline src/test/java/io/gingersnap_project/GreetingResourceTest.java:6]: This doesn't seem related :slightly_smiling_face:

File: src/main/resources/application.properties (hunk 2 above)
Comment (jabolina) [reply src/main/resources/application.properties:5]: When the test starts, it uses this properties file but runs using the test profile. Without adding the dev profile, the test will try to connect to the database/backend and fail. Looking at their docs, I didn't find a way to override the properties as a whole, only by creating a class that explicitly overrides specific keys.

Comment (jabolina) [reply src/main/resources/application.properties:5]: If using the quarkus CLI or the maven wrapper during development, everything should work the same way. The only difference is running the jar needs to pass the `Dquarkus.profile=dev`.

File: .github/workflows/pull_requests.yml (hunk 1 above)
Comment (jabolina) [reply .github/workflows/pull_requests.yml:27]: Do you see something wrong? The report action is not triggering. Maybe only triggers when the event happens in main?

Comment (ryanemerson) [reply .github/workflows/pull_requests.yml:27]: Nothing wrong, `test_report.yaml` needs to be merged for it to trigger.

File: src/main/resources/application.properties (hunk 2 above)
Comment (ryanemerson) [reply src/main/resources/application.properties:5]: Makes sense, thanks for explaining.
